File: java/src/main/java/S048.java
import com.google.common.math.LongMath;

public class S048 {

    private static final int LIMIT = 1000;
    private static final long MOD = LongMath.pow(10, 10);

    public static long solve() {
        long sum = 0;
        for (int pow = 1; pow <= LIMIT; pow++) {
            sum += modPow(pow);
        }
        return sum % LongMath.pow(10, 10);
    }

    private static long modPow(int pow) {
        long res = 1;
        for (int i = 0; i < pow; i++) {
            res  = res * pow % MOD;
        }
        return res;
    }
}

File: cpp/S048.cpp
#include <iostream>

using namespace std;

static const int LIMIT = 1000;
static const long MOD = 10L * 1000 * 1000 * 1000;

inline long n_to_n_mod(int n) {
    long res = n;
    for (int i = 1; i < n; i++) {
        res = (res * n) % MOD;
    }
    return res;
}

int main() {
    long sum = 0;
    for (int i = 1; i <= LIMIT; i++) {
        sum = (sum + n_to_n_mod(i)) % MOD;
    }
    cout << sum << endl;
    return 0;
}
